src/main/java/com/google/devtools/build/lib/remote/AbstractRemoteActionCache.java

```diff
@@ -265,7 +265,9 @@ public void onFailure(Throwable t) {
           execRoot.getRelative(file.getPath()).delete();
         }
         for (OutputDirectory directory : result.getOutputDirectoriesList()) {
-          FileSystemUtils.deleteTree(execRoot.getRelative(directory.getPath()));
+          // Only delete the directories below the output directories because the output
+          // directories will not be re-created
+          FileSystemUtils.deleteTreesBelow(execRoot.getRelative(directory.getPath()));
```

Comment:
Can you add a unit test for this change? https://github.com/bazelbuild/bazel/blob/5abfb1fbff739189269908ad23fff87b98303f6a/src/test/java/com/google/devtools/build/lib/remote/AbstractRemoteActionCacheTests.java

Reply:
Done!

Reply:
noice 👍